streaming/src/main/scala/org/apache/spark/streaming/DStreamGraph.scala

```diff
@@ -112,12 +112,10 @@ final private[streaming] class DStreamGraph extends Serializable with Logging {
 
   def generateJobs(time: Time): Seq[Job] = {
     logDebug("Generating jobs for time " + time)
-    val jobs = this.synchronized {
-      outputStreams.flatMap { outputStream =>
-        val jobOption = outputStream.generateJob(time)
-        jobOption.foreach(_.setCallSite(outputStream.creationSite))
-        jobOption
-      }
+    val jobs = getOutputStreams().flatMap { outputStream =>
```

Comment:
`synchronized` is to make sure `writeObject` never write some intermediate states of `DStreamGraph`.